pkg/common/cns-lib/vsphere/virtualcenter.go

```diff
@@ -179,9 +179,13 @@ func (vc *VirtualCenter) newClient(ctx context.Context) (*govmomi.Client, error)
 	}
 
 	s, err := client.SessionManager.UserSession(ctx)
-	if err == nil {
-		log.Infof("New session ID for '%s' = %s", s.UserName, s.Key)
+	// If session is NotAuthenticated, SessionManager.UserSession returns
+	// nil session with nil error
+	if s == nil || err != nil {
+		log.Errorf("failed to get UserSession. Session may be NotAuthenticated. err: %v", err)
+		return nil, err
```

Comment:
When session is NotAuthenticated, both "s" and "err" will be nil and this function will return.
The caller of "newClient" function check whether "err" is nil or not. If err is not nil, it will assume the client is created. However, in this case, the client actually is "nil".

Reply:
when either session is nil or we have an error we will return, else we will continue and print the session ID.

Reply:
Do you suggest anty change?

Reply:
Thanks, @lipingxue for the review. I have addressed your review comment.